### PR TITLE
sinkv2(cdc): implement EventSortEngine based on pebble

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

Issue Number: ref #5928 

### What is changed and how it works?

Add an implementation of EventSortEngine based on Pebble.

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/sorter/pebble/db.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	userProps[maxTableCRTsLabel] = fmt.Sprintf("%d", t.maxTs)
+	return nil
+}
+
+func (t *tableCRTsCollector) Name() string {
+	return tableCRTsCollectorName
+}
+
+// NOTE: both lowerBound and upperBound are included.
+func iterTable(
```
Comment (nongfushanquan) [inline pkg/sorter/pebble/db.go:66]: why are both bound included instead of one bound?

File: pkg/sorter/pebble/event_sorter.go [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+		if _, err = s.serde.Unmarshal(event, value); err != nil {
+			return
+		}
+		if s.headItem != nil {
+			break
+		}
+		s.headItem, event = event, nil
+	}
+	if s.headItem != nil {
+		if event == nil || s.headItem.CRTs != event.CRTs || s.headItem.StartTs != event.StartTs {
```
Comment (nongfushanquan) [inline pkg/sorter/pebble/event_sorter.go:256]: if event != nil  & s.headItem.CRTs == event.CTRs & s.headItem.StartTs == event.StartTs ,  what is the value of pos?

Comment (hicqu) [reply pkg/sorter/pebble/event_sorter.go:256]: In this case `pos` won't be updated. `pos` will only be filled when meets a finished transaction.

File: pkg/sorter/pebble/db.go (hunk 1 above)
Comment (hicqu) [reply pkg/sorter/pebble/db.go:66]: Generally `upperBound` comes from `resolvedTs` or `barrierTs`, which are included.
`lowerBound` can be included or excluded, I think both are OK.

**Review by nongfushanquan — APPROVED** [review]

File: pkg/sorter/pebble/event_sorter.go [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+		if err := s.cleanTable(state, tableID); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Next implements sorter.EventIterator.
+func (s *EventIter) Next() (event *model.PolymorphicEvent, pos sorter.Position, err error) {
+	var valid bool = s.iter != nil && s.iter.Valid()
```
Comment (hi-rustin) [inline pkg/sorter/pebble/event_sorter.go:242]:
```suggestion
	bool := s.iter != nil && s.iter.Valid()
```

File: pkg/sorter/pebble/event_sorter.go [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+	var value []byte = nil
```
Comment (hi-rustin) [inline pkg/sorter/pebble/event_sorter.go:243]:
```suggestion
	var value []byte
```

**Review by hi-rustin — APPROVED** [review]
rest LGTM

File: pkg/sorter/manager/pebble.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	return dbs, writeStalls, nil
+}
+
+type pebbleLogger struct{ id int }
+
+var _ pebble.Logger = (*pebbleLogger)(nil)
+
+func (logger *pebbleLogger) Infof(format string, args ...interface{}) {
+	// Do not output low-level pebble log to TiCDC log.
+	log.Debug(fmt.Sprintf(format, args...), zap.Int("db", logger.id))
```
[the remark this answers is not in the document]
Comment (hicqu) [reply pkg/sorter/manager/pebble.go:72]: DB ID can't be passed into the log if use `Debugf`.

**Comment by hi-rustin** [comment]
@hicqu Your code has some lint issues.

**Pull request merged**